File: 412-fizz-buzz/412-fizz-buzz.py
from typing import List

class Solution:
    def fizzBuzz(self, n: int) -> List[str]:
        lis = []
        for i in range(1, n+1):
            if (i % 15 == 0):
                lis.append('FizzBuzz')
            elif (i % 3 == 0):
                lis.append('Fizz')
            elif (i % 5 == 0):
                lis.append('Buzz')
            else:
                lis.append(str(i))
        return lis

File: 412-fizz-buzz/412-fizz-buzz.c
/**
 * Note: The returned array must be malloced, assume caller calls free().
 */
#include <stdlib.h>

char** fizzBuzz(int n, int* returnSize) {
    char** result = (char**) malloc(sizeof(char*) * n);
    *returnSize = n;

    for (int i = 1; i <= n; i++) {
        if (i % 3 == 0 && i % 5 == 0) {
            result[i - 1] = (char*) malloc(sizeof(char) * 9);
            strcpy(result[i - 1], "FizzBuzz");
        } else if (i % 3 == 0) {
            result[i - 1] = (char*) malloc(sizeof(char) * 5);
            strcpy(result[i - 1], "Fizz");
        } else if (i % 5 == 0) {
            result[i - 1] = (char*) malloc(sizeof(char) * 5);
            strcpy(result[i - 1], "Buzz");
        } else {
            int num_len = snprintf(NULL, 0, "%d", i);
            result[i - 1] = (char*) malloc(sizeof(char) * (num_len + 1));
            sprintf(result[i - 1], "%d", i);
        }
    }

    return result;
}